nxcomp/Log.cpp

```diff
@@ -18,50 +18,54 @@ NXLog::NXLog(char *filename)
 	level_ = NXDEBUG;
 }
 
-bool NXLog::will_log() const
+bool NXLog::will_log() 
 {
 	std::map<std::string, NXLogLevel>::const_iterator item = per_file_levels_.find(current_file());
 
 	if ( item != per_file_levels_.end() )
 	{
-		return current_level_ >= item->second;
+		return current_level() >= item->second;
 	}
 	else
 	{
-		return current_level_ >= level_;
+		return current_level() >= level();
 	}
 }
 
 
 std::string NXLogStamp::to_string() const
 {
 	static const char* level_names[] = {
-		"DBG",
-		"INF",
-		"WRN",
-		"ERR",
-		"FAT"
+		"DEBUG",
+		"INFO ",
+		"WARN ",
+		"ERROR",
+		"FATAL"
 	};
 
 	std::ostringstream oss;
 	struct tm *timeinfo;
 	char buffer[32];
 	char buffer2[32];
 
-	oss << ((level_ >=0 && level_ < NXLOG_LEVEL_COUNT ) ? level_names[level_] : "???");
+	oss << ((level() >=0 && level_ < NXLOG_LEVEL_COUNT ) ? level_names[level()] : "???");
 
 
 	timeinfo = localtime(&timestamp_.tv_sec);
 
-	if (strftime(buffer, sizeof(buffer)-1, "%Y/%m/%d %H:%M:%S", timeinfo) == 0)
-	{
-		/* contents of buffer are undefined if the data does not fit into it */
-		buffer[0] = '\0';
-	}
+	oss << std::put_time(&timeinfo, "%Y/%m/%d %H:%M:%S");
+//	if (strftime(buffer, sizeof(buffer)-1, "%Y/%m/%d %H:%M:%S", timeinfo) == 0)
+//	{
+//		/* contents of buffer are undefined if the data does not fit into it */
+//		buffer[0] = '\0';
+//	}
 
 	snprintf(buffer2, sizeof(buffer2)-1, "%03i", timestamp_.tv_usec / 1000);
 
-	oss << " " << buffer << "." << buffer2 << " " << file_ << "/" << function_ << ":" << line_ << " ";
+	oss << " " << buffer << "." << buffer2                    // Seconds and miliseconds
+	    << " " << file_ << "/" << function_ << ":" << line_   // Location
+	    << " T" << pthread_self()                             // Thread
```

Comment:
`pthread_t` could be a struct, so... that might lead to compile errors. Do we need it?